truffleruby/src/main/c/openssl/ossl_asn1.c

```diff
@@ -845,7 +845,7 @@ int_ossl_asn1_decode0_prim(unsigned char **pp, long length, long hlen, int tag,
     *pp += hlen + length;
     *num_read = hlen + length;
 
-    if (tc == sUNIVERSAL && tag < ossl_asn1_info_size && ossl_asn1_info_klass[tag]) { // ossl_asn1_info[tag].klass) {
+    if (tc == sUNIVERSAL && tag < ossl_asn1_info_size && truffle_is_truffle_object(ossl_asn1_info_klass[tag])) { // ossl_asn1_info[tag].klass) {
```

Comment:
I'm not totally sure where this is going wrong.

`ossl_asn1_info_klass` is an array allocated through our managed `malloc`. We store a C `NULL` in that array, and then get it out again, and it appears to be truthy when it shouldn't? Is that right?